concordium-consensus/src/Concordium/GlobalState/Persistent/TreeState.hs

```diff
@@ -416,12 +416,17 @@ loadSkovPersistentData rp _treeStateDirectory pbsc = do
                 "The block state database is corrupt. Recovery attempt failed: " <> e
         Right (_lastFinalizationRecord, lfStoredBlock) -> do
             -- Truncate the blobstore beyond the last finalized blockstate.
-            liftIO $ truncateBlobStore (bscBlobStore . PBS.pbscBlobStore $ pbsc) (sbState lfStoredBlock)
+            liftIO $ truncateBlobStore (bscBlobStore . PBS.pbscBlobStore $ pbsc) (sbState . fst $ lfStoredBlock)
             -- Get the genesis block.
             genStoredBlock <-
                 maybe (logExceptionAndThrowTS GenesisBlockNotInDataBaseError) return
                     =<< liftIO (getFirstBlock _db)
             _genesisBlockPointer <- liftIO $ makeBlockPointer genStoredBlock
+            when (isNothing (snd genStoredBlock)) $ do
```

Comment:
It'd be useful to clarify here that this is only expected to run when upgrading from node version < 6.1